beetsplug/alternatives.py

```diff
@@ -166,29 +197,50 @@ def parse_config(self, config):
             dir = os.path.join(self.lib.directory, dir)
         self.directory = dir
 
+    def is_in_external(self, path):
+        return os.path.isfile(syspath(path))
+
+    def item_change_actions(self, item, path, dest):
+        """ Returns the necessary actions for items that were previously in the
+        external collection, but might require metadata updates.
+        """
+        actions = []
+
+        if not util.samefile(path, dest):
+            actions.append(self.MOVE)
+
+        item_mtime_alt = os.path.getmtime(syspath(path))
+        if (item_mtime_alt < os.path.getmtime(syspath(item.path))):
+            actions.append(self.WRITE)
+        album = item.get_album()
+
+        if album:
+            if (album.artpath and
+                    os.path.isfile(syspath(album.artpath)) and
+                    (item_mtime_alt
+                     < os.path.getmtime(syspath(album.artpath)))):
+                actions.append(self.SYNC_ART)
+
+        return actions
+
     def matched_item_action(self, item):
         path = self.get_path(item)
-        if path and os.path.isfile(syspath(path)):
+        print_("get_path(item): ")
+        try:
+            print_(repr(path))
+            if path:
+                    print_("lstat(get_path(item)): " + str(os.lstat(syspath(path))))
+        except:
+            pass
+        if path and self.is_in_external(path):
```

Comment:
We could also do the following:
```suggestion
        if path and os.path.lexists(syspath(path)):
```
This would work for real and linked external views and reduce the indirection and number of methods that are different between the classes.

Reply:
Right, that should do the trick :)